src/module.windows.ts

```diff
@@ -0,0 +1,71 @@
+import {NativeModules} from 'react-native';
+import {RESULTS, PERMISSIONS} from './constants';
+import {Contract} from './contract';
+import {NotificationsResponse, Permission, PermissionStatus} from './types';
+import {uniq} from './utils';
+
+const RNP: {
+  OpenSettings: () => Promise<void>;
+  CheckNotifications: () => Promise<PermissionStatus>;
+  Check: (permission: Permission) => Promise<PermissionStatus>;
+  Request: (permission: Permission) => Promise<PermissionStatus>;
+} = NativeModules.RNPermissions;
+
+async function openSettings(): Promise<void> {
+  return RNP.OpenSettings();
+}
+
+async function check(permission: Permission): Promise<PermissionStatus> {
+  return RNP.Check(permission);
+}
+
+async function request(permission: Permission): Promise<PermissionStatus> {
+  return RNP.Request(permission);
+}
+
+async function checkNotifications(): Promise<NotificationsResponse> {
+  const status = await RNP.CheckNotifications();
+  return {status, settings: {}};
+}
+
+async function requestNotifications(): Promise<NotificationsResponse> {
+  // There is no way to request notifications on Windows if they are
+  // disabled.
+  return checkNotifications();
+}
+
+async function checkMultiple<P extends Permission[]>(
+  permissions: P,
+): Promise<Record<P[number], PermissionStatus>> {
+  const result = {} as Record<P[number], PermissionStatus>;
+  const promises = permissions.map(async (permission: P[number]) => {
+    const promise = check(permission);
+    result[permission] = await promise;
+    return promise;
+  });
+  Promise.all(promises);
+  return result;
+}
+
+async function requestMultiple<P extends Permission[]>(
+  permissions: P,
+): Promise<Record<P[number], PermissionStatus>> {
+  const result = {} as Record<P[number], PermissionStatus>;
+  const promises = permissions.map(async (permission: P[number]) => {
+    const promise = request(permission);
+    result[permission] = await promise;
+    return promise;
+  });
+  Promise.all(promises);
```

Comment:
Same except you have to run the requests in sequence (and not in parallel) as the user receive one request at the time:

```ts
async function requestMultiple<P extends Permission[]>(
  permissions: P,
): Promise<Record<P[number], PermissionStatus>> {
  type Output = Record<P[number], PermissionStatus>;

  const output: Partial<Output> = {};
  const dedup = uniq(permissions);

  for (let index = 0; index < dedup.length; index++) {
    const permission: P[number] = dedup[index];
    output[permission] = await request(permission);
  }

  return output as Output;
}
```

Reply:
<img width="1952" alt="Screenshot 2020-10-18 at 23 13 49" src="https://user-images.githubusercontent.com/1902323/96386232-87be5b80-1199-11eb-9bc6-e41d60d53cb8.png">

`result` is logged using `console.warn`, as you can see, it does not wait.